src/ucp/rndv/rndv.c

```diff
@@ -1513,9 +1504,22 @@ ucs_status_t ucp_rndv_rts_handle_status_from_pending(ucp_request_t *sreq,
         }
 
         ucp_send_request_id_release(sreq);
-        ucp_request_complete_and_dereg_send(sreq, status);
+
+        if (sreq->flags & UCP_REQUEST_FLAG_SEND_TAG) {
+            /* RTS: the send request is controlled by a user */
+            ucp_request_complete_and_dereg_send(sreq, status);
+        } else {
+            /* RTR: the send request is derived request of the receive request
+             * which is controlled by a user */
+            super_req = ucp_request_get_super(sreq);
```

Comment:
super_req -> recv_req
BTW, does it work for frag-RTR?

Reply:
> BTW, does it work for frag-RTR?

no, forgot to handle it too.
done.

> super_req -> recv_req

done, but used `rreq` instead of `recv_req` to align with `sreq`